### PR TITLE
Code Review

### PR DESCRIPTION
**Pull request by Urchmaney**

### CODE REVIEW

**Comment by bellom** [comment]
currently reviewing this PR

**Review by bellom — CHANGES_REQUESTED** [review]
Hi @Urchmaney @fatymahmed,
Nice work using the `GitHub Flow`, kindly be aware that you are to use `Git Flow` in the next project.
Short Explanation Hint: 
`GitHub Flow`: Pull Requests is raise from feature branches to the master.
`Git Flow`: Pull Requests is raise from feature branches to the development.
I tested the `bubble_sort_by` method, it gave wrong output at my end....
output > 0..3
expected output > ["hi","hey", "hello"]

Kindly fix the mentioned issues, fix linter and re-submit for review.

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+def bubble_sort(arr)  
+  arr.each { |element|
+    arr.each_with_index { |element,index|
+    unless index == arr.length - 1
+    if arr[index] > arr[index + 1]        
+      tmp = arr[index]
+      arr[index] = arr[index + 1]
+      arr[index+1] = tmp
```
Comment (bellom) [inline bubble_sort.rb]: Kindly know that one the Microverse guidelines encourage students to leave space around operators [read more](https://github.com/github/rubocop-github/blob/master/STYLEGUIDE.md#ruby-style-guide)
And you can see the linter also caught that......kindly fix it.

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    end
+    end
+    }
+  }
+end
+
+def bubble_sort_by (array)
+  for i in 0...array.length
+  for j in 0...array.length - 1 
+    check=yield(array[j],array[j + 1])
```
Comment (bellom) [inline bubble_sort.rb]: Kindly know that one the Microverse guidelines encourage students to leave space around operators [read more](https://github.com/github/rubocop-github/blob/master/STYLEGUIDE.md#ruby-style-guide)
And you can see the linter also caught that......kindly fix it.

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    if check > 0
+    temp = array[j]
+    array[j]=array[j + 1]
```
Comment (bellom) [inline bubble_sort.rb]: Kindly know that one the Microverse guidelines encourage students to leave space around operators [read more](https://github.com/github/rubocop-github/blob/master/STYLEGUIDE.md#ruby-style-guide)
And you can see the linter also caught that......kindly fix it.

**Review by bellom — CHANGES_REQUESTED** [review]

File: bubble_sort.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    if arr[index] > arr[index + 1]        
+      tmp = arr[index]
+      arr[index] = arr[index + 1]
+      arr[index + 1] = tmp
+    end
+    end
+    }
+  }
+end
+
```
Comment (mosaaleb) [inline bubble_sort.rb:16]: Please remove those curly braces, as the general convention of multi-line block is to use `do ... end`. Curly braces used in one liner blocks. [curly braces vs do end](https://stackoverflow.com/questions/5587264/do-end-vs-curly-braces-for-blocks-in-ruby)

File: bubble_sort.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+def bubble_sort_by(array)
+  for i in 0...array.length
+  for j in 0...array.length - 1 
+    check = yield(array[j],array[j + 1])
+    if check > 0
+    temp = array[j]
```
Comment (mosaaleb) [inline bubble_sort.rb]: You can put lines 18 and 19 in the same line, without the need for extra variable `if yield(array[j], array[j + 1]) > 0`

**Review by mosaaleb — CHANGES_REQUESTED** [review]
#### Well done@fatymahmed @Urchmaney 

The logic is right and the test cases have passed :clap:. But you still have a lot of problems in your code as mentioned by my fellow TSE related to styling guidelines.

Screenshot from 2019-08-07 22-40-45

- [ ] I see that you have `.rubocop.yml` in your repo. Please fix all related problems indicated by rubocop plugin.

- [ ] Please, see my comments below.

Kindly handle these issues and submit a new code review.

### Happy Coding :computer:

**Comment by bellom** [comment]
## Currently reviewing this PR

**Review by bellom — APPROVED** [review]
Hi @Urchmaney @fatymahmed,
Nice work using the `GitHub Flow`, kindly be aware that you are to use `Git Flow` in the next project.
Good job fixing all the necessary errors from the previous reviews.
## Congratulation Guys :+1:

**Pull request merged**